improver_tests/cube_combiner/test_CubeCombiner.py

```diff
@@ -237,6 +255,51 @@ def test_exception_for_single_entry_cubelist(self):
         with self.assertRaisesRegex(ValueError, msg):
             plugin.process(cubelist, "new_cube_name")
 
+    def test_broadcast_coord(self):
+        """Test that plugin broadcasts to a coord"""
+        plugin = CubeCombiner("*")
+        cube = self.cube4[:, 0, ...].copy()
+        cube.data = np.ones_like(cube.data)
+        cube.remove_coord("lwe_thickness_of_precipitation_amount")
+        cubelist = iris.cube.CubeList([self.cube4.copy(), cube])
+        result = plugin.process(
+            cubelist, "new_cube_name", broadcast_to_coords=["threshold"]
```

Comment:
It looks like, based on the output for this unit test, that the input `cube` is modified by this plugin. Is it desirable for this plugin to result in changes to the input cubes?

Reply:
I can't see what you're getting at here. Why do you think the input cube is being modified? I have run a quick test to show that it isn't (at least, not the data) - I can expand this to meta-data and include it if you wish.

Reply:
The difference between `cube` before and after running this unit test is the presence of a scalar `lwe_thickness_of_precipitation_amount` coordinate (please see the cube printouts below). This is arising from the line `cube.add_aux_coord(new_coord, None)` within `_setup_coords_for_broadcast` where the `cube` is being modified in-place. Should the `cube` in `_setup_coords_for_broadcast` be copied prior to adding this auxiliary coordinate, to avoid this side effect?

`cube` before
```
probability_of_lwe_thickness_of_precipitation_amount_above_threshold / (1) (realization: 3; latitude: 2; longitude: 2)
     Dimension coordinates:
          realization                                                                  x            -             -
          latitude                                                                     -            x             -
          longitude                                                                    -            -             x
     Scalar coordinates:
          forecast_period: 10800 seconds, bound=(7200, 10800) seconds
          forecast_reference_time: 2015-11-18 22:00:00
          time: 2015-11-19 01:00:00, bound=(2015-11-19 00:00:00, 2015-11-19 01:00:00)
```

`cube` after:
```
probability_of_lwe_thickness_of_precipitation_amount_above_threshold / (1) (realization: 3; latitude: 2; longitude: 2)
     Dimension coordinates:
          realization                                                                  x            -             -
          latitude                                                                     -            x             -
          longitude                                                                    -            -             x
     Scalar coordinates:
          forecast_period: 10800 seconds, bound=(7200, 10800) seconds
          forecast_reference_time: 2015-11-18 22:00:00
          lwe_thickness_of_precipitation_amount: 0 K
          time: 2015-11-19 01:00:00, bound=(2015-11-19 00:00:00, 2015-11-19 01:00:00)
```

Reply:
Ah yes, I hadn't spotted that. I've added a `cube.copy()` to prevent this. I thought this was worth testing, so I added a simple test for this too.